src/core/ucc_lib.c

```diff
@@ -18,3 +28,77 @@ ucc_status_t ucc_init_version(unsigned api_major_version,
 {
     return UCC_ERR_NOT_IMPLEMENTED;
 }
+
+ucc_status_t ucc_lib_config_read(const char *env_prefix, const char *filename,
+                                 ucc_lib_config_t **config_p)
+{
+    ucc_lib_config_t *config;
+    ucc_status_t      status;
+    int               full_prefix_len;
+    const char       *base_prefix = "UCC_";
+
+    config = ucc_malloc(sizeof(*config), "lib_config");
+    if (config == NULL) {
+        ucc_error("failed to allocate %zd bytes for lib_config",
+                  sizeof(*config));
+        status = UCC_ERR_NO_MEMORY;
+        goto err;
+    }
+    /* full_prefix for a UCC lib config is either just
+          (i)  "UCC_" - if not "env_prefix" is provided, or
+          (ii) "AAA_UCC" - where AAA is the value of "env_prefix".
+       Need to allocate enough space to build the proper prefix str: 2 more
+       chars for "_" and "\0". */
+    full_prefix_len =
+        strlen(base_prefix) + (env_prefix ? strlen(env_prefix) : 0) + 2;
+    config->full_prefix = ucc_malloc(full_prefix_len, "full_prefix");
+    if (!config->full_prefix) {
+        ucc_error("failed to allocate %zd bytes for full_prefix",
+                  full_prefix_len);
+        status = UCC_ERR_NO_MEMORY;
+        goto err_free_config;
+    }
+    if (env_prefix) {
+        snprintf(config->full_prefix, full_prefix_len, "%s_%s", env_prefix,
+                 base_prefix);
+    } else {
+        strcpy(config->full_prefix, base_prefix);
```

Comment:
Though it is correct here, should we avoid using strcpy and instead use snprintf ? Also, should we use ucs_snprintf_zero eventually ?

Reply:
@manjugv  @Sergei-Lebedev  changed strcpy to ucc_strncpy_safe. Regarding snprintf: i've added ucc_snprintf_safe. However, the implementation in UCS (ucs_snprintf_safe) turned to have a bug - it truncates last char (i've reported that to UCX people). Hence, so far i've defined ucc_snprintf_safe to just snprintf from stdlib (which does guarantee '\0' appended btw).